session/session_test.go

```diff
@@ -48,11 +51,55 @@ func TestReceiveEventWhenSubscribing(t *testing.T) {
 		if event != SessionDisconnected {
 			t.Error("Expected to receive disconnected: ", event)
 		}
+	case <-time.After(5 * time.Second):
+		t.Error("Failed to receive event")
+	}
 
-		if <-events != SessionReconnected {
+	select {
+	case event := <-events:
+		if event != SessionReconnected {
 			t.Error("Expected to receive reconnected: ", event)
 		}
-	case <-time.After(100 * time.Millisecond):
+	case <-time.After(5 * time.Second):
 		t.Error("Failed to receive event")
 	}
 }
+
+func TestResumeZKSessionWithValidSession(t *testing.T) {
+	proxy := CreateProxy(t)
+	defer proxy.Delete()
+
+	store, err := NewZKSession("localhost:27445", 200*time.Millisecond, nil)
+	if err != nil {
+		t.Error("Failed to connect to Zookeeper: ", err)
+	}
+	defer store.Close()
+
+	clientId := store.ClientId()
+	clientId.Save()
```

Comment:
The `Save()` here is unnecessary. Scott's patch (I had difficulty understanding this at first) makes you able to dump the client id to a file and recover it. At runtime, there are no problems (as this test proves). This test (and your code) would pass without Scott's patch. This test by itself is useful because it tests runtime resumption of a session (although I think this that is handled automatically).

I would consider adding another test that does `serializedClientId := clientId.Save()` and then does a `LoadClientId` based on that. Serializing and deserializing is how this patch will be used in follow-ups internally.